hwtypes/adt.py

```diff
@@ -77,6 +77,11 @@ def value_dict(self):
             d[k] = getattr(self, k)
         return MappingProxyType(d)
 
+
+class Product(AnonymousProduct, metaclass=ProductMeta):
+    def __repr__(self):
+        return f'{type(self).__name__}({", ".join(f"{k}={v}" for k,v in self.value_dict.items())})'
```

Comment:
Does this need to be explicitly overridden to get the right name? Seems identical to the above implementation for `AnonymousProduct` (maybe I'm missing a subtle difference)

Reply:
No, it does not.  Will remove.